pkgs/applications/misc/navi/default.nix

```diff
@@ -0,0 +1,29 @@
+{ rustPlatform, fetchFromGitHub, lib, fzf, makeWrapper }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "navi";
+  version = "2.7.1";
+
+  src = fetchFromGitHub {
+    owner = "denisidoro";
+    repo = "navi";
+    rev = "v${version}";
+    sha256 = "12p9l41k7isaapr0xbsm7brkjrv7i8826y029i12psz92nsynk29";
+  };
+
+  cargoSha256 = "11dc3gc7fyikbbgacmljhysr2sl7lmq6w3bsfcf2cqny39r25yp0";
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  postInstall = ''
```

Comment:
Upstream also provides shell completion files:

https://github.com/denisidoro/navi/tree/master/shell

You install install them using `installShellCompletion` after putting `installShellFiles` in `nativeBuildInputs`.

Reply:
They are not really completions, they are adding key binding for navi widget, and upstream recommends [opting into it via invoking subcommand](https://github.com/denisidoro/navi#shell-widget) — would we want to implicitly set it up for users?

Reply:
Ah, sorry for the noise. No, I don't think we should set them up implicitly.

Reply:
No worries, thank you for review!